lib/helpers/postDetails.ts

```diff
@@ -0,0 +1,22 @@
+export function getPostDetails(post: any) {
+  let icon = "";
+  let backgroundColor = "";
+  let textColor = "";
+  if (post.type === "workflow") {
+    icon = "/icons/workflow.svg";
+    backgroundColor = "bg-primary-900";
+    textColor = "text-primary-500";
+  }
+  if (post.type === "component") {
+    icon = "/icons/component.svg";
+    backgroundColor = "bg-myPurple-900";
+    textColor = "text-myPurple-500";
+  }
+  if (post.type === "knowledge") {
+    icon = "/icons/knowledge.svg";
+    backgroundColor = "bg-myGreen-900";
+    textColor = "text-myGreen-500";
+  }
+
+  return { icon, backgroundColor, textColor };
+}
```

Comment:
Not entirely sure about the naming here. Is this function just meant to turn the post type into the data for our  filterPill?

Maybe we can rename this as it doesn't get any postDetails
```suggestion
export function getPostDetails(post: any) {
  switch (post.type) {
    case "workflow":
      return {
        icon: "/icons/workflow.svg",
        backgroundColor: "bg-primary-900",
        textColor: "text-primary-500",
      };
    case "component":
      return {
        icon: "/icons/component.svg",
        backgroundColor: "bg-myPurple-900",
        textColor: "text-myPurple-500",
      };
    case "knowledge":
      return {
        icon: "/icons/knowledge.svg",
        backgroundColor: "bg-myGreen-900",
        textColor: "text-myGreen-500",
      };
    default:
      return {
        icon: "",
        backgroundColor: "",
        textColor: "",
      };
  }
}

```

Maybe instead of using 'let' and assigning variables we could either have our if-statements return the object directly or we can use the slightly faster switch statement to return this information.